From e6d5c8fd29512bb1fc25f2094d7f5884573bf673 Mon Sep 17 00:00:00 2001
From: Ewan Milne <emilne@redhat.com>
Date: Wed, 20 Aug 2014 18:58:53 -0400
Subject: [scsi] export device_busy for sdev

Message-id: <1408561176-18133-15-git-send-email-emilne@redhat.com>
Patchwork-id: 88425
O-Subject: [RHEL7.1 PATCH e-stor 14/57] [SCSI] export device_busy for sdev
Bugzilla: 1132092
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: "Ewan D. Milne" <emilne@redhat.com>

upstream commit 522db3c9e1a5d3e9bfbf23d9106180651338d1bd

Tested by Ewan D. Milne and Barry Donahue with a combination
of manual and automated testing.  No issues seen.

upstream commit 522db3c9e1a5d3e9bfbf23d9106180651338d1bd
Author: Jack Wang <jinpu.wang@profitbricks.com>
Date:   Thu Sep 12 17:29:52 2013 +0200

    [SCSI] export device_busy for sdev

    If you mutiple devices connect to a host, we might be interested in
    have an intensive I/O workload on one disk, and notice starvation on others.
    This give the user more hint about current infight io for scsi device.

    Signed-off-by: Jack Wang <jinpu.wang@profitbricks.com>
    Signed-off-by: James Bottomley <JBottomley@Parallels.com>

Signed-off-by: Ewan D. Milne <emilne@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/scsi_sysfs.c b/drivers/scsi/scsi_sysfs.c
index 9cd8912..9117d0b 100644
--- a/drivers/scsi/scsi_sysfs.c
+++ b/drivers/scsi/scsi_sysfs.c
@@ -580,6 +580,7 @@ static int scsi_sdev_check_buf_bit(const char *buf)
  */
 sdev_rd_attr (device_blocked, "%d\n");
 sdev_rd_attr (queue_depth, "%d\n");
+sdev_rd_attr (device_busy, "%d\n");
 sdev_rd_attr (type, "%d\n");
 sdev_rd_attr (scsi_level, "%d\n");
 sdev_rd_attr (vendor, "%.8s\n");
@@ -801,6 +802,7 @@ static struct attribute *scsi_sdev_attrs[] = {
  &dev_attr_device_blocked.attr,
  &dev_attr_type.attr,
  &dev_attr_scsi_level.attr,
+ &dev_attr_device_busy.attr,
  &dev_attr_vendor.attr,
  &dev_attr_model.attr,
  &dev_attr_rev.attr,
-- 
1.7.1